src/components/ProbeStatus.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useState } from 'react';
+import { css } from 'emotion';
+import { Badge, BadgeColor, Button, Container, ConfirmModal, Legend, IconName } from '@grafana/ui';
+import { Probe, OrgRole } from 'types';
+import { SMDataSource } from 'datasource/DataSource';
+import { hasRole } from 'utils';
+import { UptimeGauge } from './UptimeGauge';
+
+interface Props {
+  probe: Probe;
+  instance: SMDataSource;
+  onResetToken: () => void;
+}
+
+interface BadgeStatus {
+  color: BadgeColor;
+  text: string;
+  icon: IconName;
+}
+
+const getBadgeStatus = (online: boolean): BadgeStatus => {
+  if (online) {
+    return {
+      text: 'Online',
+      color: 'green',
+      icon: 'heart',
+    };
+  }
+  return {
+    text: 'Offline',
+    color: 'red',
+    icon: 'heart-break',
+  };
+};
+
+const ProbeStatus: FC<Props> = ({ probe, instance, onResetToken }) => {
+  const [showResetModal, setShowResetModal] = useState(false);
+
+  if (!probe) {
+    return null;
+  }
+  const isEditor = !probe.public && hasRole(OrgRole.EDITOR);
+  const badgeStatus = getBadgeStatus(probe.online);
+
+  const handleResetToken = () => {
+    onResetToken();
+    setShowResetModal(false);
+  };
+
+  return (
+    <Container margin="md">
+      <div
+        className={css`
```

Comment:
maybe consider this approach:
https://github.com/grafana/grafana/blob/master/contribute/style-guides/styling.md#styling-with-theme